### PR TITLE
MAINTAINERS: update hal_silabs maintainers

### PR DESCRIPTION
**Pull request by fkokosinski**
This PR updates the maintainers list (adds @sateeshkotapati) for the hal_silabs area and changes its status to `maintained`.

### CODE REVIEW

**Review by stephanosio — CHANGES_REQUESTED** [review]

File: MAINTAINERS.yml
Comment (fkokosinski) [reply MAINTAINERS.yml:2333]: Hi Stephanos

There is an ongoing effort to add new SiLabs platforms to Zephyr [1]. It is currently partially blocked by the lack of reviews of the PR to the `hal_silabs` repository [2]. This PR also doesn't have any assignee, because the `west project: hal_silabs` area doesn't have a maintainer assigned to it. SiLabs has selected Sateesh (@sateeshkotapati) as a potential maintainer of their HAL repository in the Zephyr project. He had previously spearheaded their effort of adding new SiLabs platforms [3][4], so we think that he's a good fit for being responsible for the SiLabs HAL in Zephyr, while we help him and SiLabs follow all the relevant policies of the Zephyr Project.

1: https://github.com/zephyrproject-rtos/zephyr/pull/50352
2: https://github.com/zephyrproject-rtos/hal_silabs/pull/19
3: https://github.com/zephyrproject-rtos/zephyr/pull/35361
4: https://github.com/zephyrproject-rtos/hal_silabs/pull/13

Comment (stephanosio) [reply MAINTAINERS.yml:2333]: Ok. If @sateeshkotapati works for SiLabs, then it makes a bit more sense; though, this is not a normal/documented process in https://docs.zephyrproject.org/latest/project/project_roles.html.

@nashif Should this be brought up and voted at the TSC per the documented process?

**Comment by fkokosinski** [comment]
Hi @stephanosio

Taking the feedback above into consideration, we wanted to propose a different way to approach the topic of `hal_silabs` maintenance. This means moving @sateeshkotapati to the `collaborator` tier and adding myself as a maintainer of this HAL. Doing so would let us get this HAL maintained confirming to the process, and over time Sateesh could take over in the maintainer role.

Let me know if you think this solution would be more acceptable.

**Review by stephanosio — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**
